teos/cli/teos_cli.py

```diff
@@ -209,18 +343,22 @@ def run():
 
             if opt in ["-h", "--help"]:
                 sys.exit(show_usage())
-
-        command = args.pop(0) if args else None
-        if command in commands:
-            main(command, args, data_dir, command_line_conf)
-        elif not command:
-            sys.exit("No command provided. Use help to check the list of available commands")
-        else:
-            sys.exit("Unknown command. Use help to check the list of available commands")
-
     except GetoptError as e:
         sys.exit("{}".format(e))
 
+    if command in CLI.COMMANDS:
+        try:
+            cli = CLI(data_dir, command_line_conf)
+            result = cli.run(command, command_args)
+            if result:
+                print(result)
+        except InvalidParameter:
+            sys.exit(show_usage())
```

Comment:
This should first print the exception message, and then `sys.exit` Since we are casting everything to BasicException (and InvalidParameter inherits from it), here we could simply do:

```
print(e.msg if not e.kwargs else f"{e.msg}. Error arguments: {e.kwargs}")
sys.exit(show_usage())
```

Reply:
Done in 9e4b087, although this is pending some other changes as per comment below.

Reply:
Replied bellow, this may need rebase then.